account_multicurrency_revaluation/model/account.py

```diff
@@ -54,83 +78,109 @@ def _onchange_user_type_id(self):
-            ") SELECT account_id as id, partner_id, currency_id, "
+            """
+WITH amount AS (
+    SELECT
+        aml.account_id,
+        CASE WHEN aat.type IN ('payable', 'receivable')
+            THEN aml.partner_id
+            ELSE NULL
+        END AS partner_id,
+        aml.currency_id,
+        aml.debit,
+        aml.credit,
+        aml.amount_currency
+    FROM account_move_line aml
+    LEFT JOIN account_move am ON aml.move_id = am.id
+    INNER JOIN account_account acc ON aml.account_id = acc.id
+    INNER JOIN account_account_type aat ON acc.user_type_id = aat.id
+    LEFT JOIN account_partial_reconcile aprc
+        ON (aml.balance < 0 AND aml.id = aprc.credit_move_id)
+    LEFT JOIN account_move_line amlcf
+        ON (
+            aml.balance < 0
+            AND aprc.debit_move_id = amlcf.id
+            AND amlcf.date < %s
+        )
+    LEFT JOIN account_partial_reconcile aprd
+        ON (aml.balance > 0 AND aml.id = aprd.debit_move_id)
+    LEFT JOIN account_move_line amldf
+        ON (
+            aml.balance > 0
+            AND aprd.credit_move_id = amldf.id
+            AND amldf.date < %s
+        )
+    WHERE
+        aml.account_id IN %s
+        AND aml.date <= %s
+        """
+            + (("AND aml.date >= %s") if start_date else "")
+            + """
+        AND aml.currency_id IS NOT NULL
+        AND am.state = 'posted'
+    GROUP BY
+        aat.type,
+        aml.id
+    HAVING
+        aml.full_reconcile_id IS NULL
+        OR (MAX(amldf.id) IS NULL AND MAX(amlcf.id) IS NULL)
+)
+SELECT
+    account_id as id,
+    partner_id,
+    currency_id,
+"""
             + ", ".join(self._sql_mapping.values())
-            + " FROM amount "
-            + (("WHERE " + where_clause) if where_clause else " ")
-            + " GROUP BY account_id, currency_id, partner_id"
+            + """
+FROM amount
+"""
+            + (("WHERE " + where_clause) if where_clause else "")
+            + """
+GROUP BY account_id, currency_id, partner_id
+        """
         )
 
-        params = []
-        params.append(revaluation_date)
-        params.append(revaluation_date)
-        params.append(tuple(self.ids))
-        params.append(revaluation_date)
-        params += where_clause_params
+        params = [
+            revaluation_date,
+            revaluation_date,
+            tuple(self.ids),
+            revaluation_date,
+            *where_clause_params,
+        ]
+        if start_date:
+            # Insert the value after the revaluation date parameter
+            params.insert(4, start_date)
+
         return query, params
 
-    def compute_revaluations(self, revaluation_date):
-        accounts = {}
-        # compute for each account the balance/debit/credit from the move lines
-        query, params = self._revaluation_query(revaluation_date)
+    def compute_revaluations(self, start_date, revaluation_date):
+        query, params = self._revaluation_query(start_date, revaluation_date)
```

Comment:
Especially for public method like this.
```suggestion
    def compute_revaluations(self, revaluation_date, start_date=None):
        query, params = self._revaluation_query(revaluation_date, start_date)
```